### PR TITLE
Internationalisation: Translate the Share modal 

### LINKED ISSUE #52200: Translate the Share modal
In the process of getting ready for the translation of the portal, we need to extract strings into translation tags. One of our first steps is to translate the Dashboard in view only mode. This ticket does part of that by translating the share modal.

### PR DESCRIPTION
**Pull request by eledobleefe**
**What this PR does / why we need it**:
In the process of getting ready for the translation of the portal, we need to extract strings into translation tags. One of our first steps is to translate the Dashboard in view-only mode. This ticket does part of that by translating the share modal.

**Which issue(s) this PR fixes**:

Fixes #52200 

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/dashboard/components/ShareModal/ShareLink.tsx [hunk 1]
```diff
@@ -91,37 +86,86 @@ export class ShareLink extends PureComponent<Props, State> {
...
+          <Field label={linkURLTranslation}>
             <Input
               id="link-url-input"
               value={shareUrl}
               readOnly
               addonAfter={
                 <ClipboardButton variant="primary" getText={this.getShareUrl} onClipboardCopy={this.onShareUrlCopy}>
-                  <Icon name="copy" /> Copy
+                  <Icon name="copy" />
+                  <Trans id="share-modal.link.copy-link-button">Copy</Trans>
```
Comment (JoaoSilvaGrafana) [inline public/app/features/dashboard/components/ShareModal/ShareLink.tsx:168]: I think we lost the space between the icon and "Copy" here

File: public/app/features/dashboard/components/ShareModal/ShareLink.tsx [hunk 2]
```diff
@@ -133,31 +177,44 @@ export class ShareLink extends PureComponent<Props, State> {
             {isDashboardSaved && (
               <div className="gf-form">
                 <a href={imageUrl} target="_blank" rel="noreferrer" aria-label={selectors.linkToRenderedImage}>
-                  <Icon name="camera" /> Direct link rendered image
+                  <Icon name="camera" />
+                  <Trans id="share-modal.link.rendered-image">Direct link rendered image</Trans>
```
Comment (JoaoSilvaGrafana) [inline public/app/features/dashboard/components/ShareModal/ShareLink.tsx:181]: And here

File: public/app/features/dashboard/components/ShareModal/ShareLink.tsx [hunk 3]
```diff
@@ -133,31 +177,44 @@ export class ShareLink extends PureComponent<Props, State> {
...
-              Grafana image renderer plugin
-            </a>
-            . Please contact your Grafana administrator to install the plugin.
+          <Alert
+            severity="info"
+            title={t({ id: 'share-modal.link.render-alert', message: `Image renderer plugin not installed` })}
+            bottomSpacing={0}
+          >
+            <Trans id="share-modal.link.render-instructions">
+              To render a panel image, you must install the &nbsp;
```
Comment (JoaoSilvaGrafana) [inline public/app/features/dashboard/components/ShareModal/ShareLink.tsx:207]: There are two spaces at the end of this sentence, remove the one before `&nbsp;`

File: public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx [hunk 4]
```diff
@@ -207,50 +232,71 @@ export class ShareSnapshot extends PureComponent<Props, State> {
...
-            metrics."
-        >
+        <Field label={timeoutTranslation} description={timeoutDescriptionTranslation}>
           <Input id="timeout-input" type="number" width={21} value={timeoutSeconds} onChange={this.onTimeoutChange} />
         </Field>
 
         <Modal.ButtonRow>
           <Button variant="secondary" onClick={onDismiss} fill="outline">
-            Cancel
+            <Trans id="share-modal.snapshot.copy-button">Cancel</Trans>
```
Comment (JoaoSilvaGrafana) [inline public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx:291]: The id here should be `cancel-button`

File: public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx [hunk 5]
```diff
@@ -207,50 +232,71 @@ export class ShareSnapshot extends PureComponent<Props, State> {
...
           </Button>
           {externalEnabled && (
             <Button variant="secondary" disabled={isLoading} onClick={this.createSnapshot(true)}>
-              {sharingButtonText}
+              <Trans id="share-modal.snapshot.sharing-button">{sharingButtonText}</Trans>
```
Comment (JoaoSilvaGrafana) [inline public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx:295]: The translators can't translate this 😅  just remove the `Trans` component, do we know if we have to translate this from somewhere else?

File: public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx [hunk 6]
```diff
@@ -269,15 +315,15 @@ export class ShareSnapshot extends PureComponent<Props, State> {
...
             <ClipboardButton variant="secondary" getText={this.getSnapshotUrl} onClipboardCopy={this.onSnapshotUrlCopy}>
-              Copy Link
+              <Trans id="share-modal.snapshot.copy-link-button">Copy Link</Trans>
             </ClipboardButton>
           </div>
         </div>
 
         <div className="pull-right" style={{ padding: '5px' }}>
-          Did you make a mistake?{' '}
+          <Trans id="share-modal.snapshot.mistake-message">Did you make a mistake? </Trans>
```
Comment (JoaoSilvaGrafana) [inline public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx:324]: Space is missing after this message

**Review by JoaoSilvaGrafana — COMMENTED** [review]
This looks really good 🎉 
Some minor comments and I noticed one thing that is missing translate, when you are in the export tab and press "View JSON" it opens a new modal of the component `ViewJsonModal` and that has 1 more sentence to translate 😄

File: public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx (hunk 4 above)
Comment (eledobleefe) [reply public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx]: Yes, you're right @JoaoSilvaGrafana

**Review by JoaoSilvaGrafana — APPROVED** [review]
Great job Laura 🎉

**Pull request merged**